<launch>
  <arg name="width"  default="1280" />
  <arg name="height" default="960" />
  <arg name="fps"    default="54" />
  <arg name="namespace" default="camera" />

  <arg name="vendor" default="0x2560" />
  <arg name="product" default="0xc110" />
  <arg name="serial"  default="36234D0B"/>
  <arg name="node_name" default="left_master" />
  <arg name="nodelet_manager" default="/manager" />

  <arg name="video_mode" default="uncompressed" />
  <arg name="camera_info_url" default="file://$(env ROS_HOME)/camera_info/cameraLeft.yaml"/>
  <arg name="timestamp_method" default="start"/>
  <arg name="auto_exposure" default="3"/>
  <arg name="auto_white_balance" default="false" />
  <arg name="auto_focus"         default="false" />
  <arg name="exposure_absolute" default="0.0030"/>
  <arg name="scanning_mode" default="0"/>
  <arg name="auto_exposure_priority" default="0"/>
  <arg name="iris_absolute" default="0.0"/>

  <node pkg="nodelet" type="nodelet" name="$(arg node_name)" args="load libuvc_camera/driver $(arg nodelet_manager)" output="screen" >
    <remap from="image_raw" to="$(arg node_name)/image_raw"/>
    <remap from="set_camera_info" to="$(arg node_name)/set_camera_info"/>
    <!-- Parameters used to find the camera -->
    <param name="vendor" value="$(arg vendor)"/>
    <param name="product" value="$(arg product)"/>
    <param name="serial" value="$(arg serial)"/>
    <!-- If the above parameters aren't unique, choose the first match: -->
    <param name="index" value="0"/>
    
    <!-- Image size and type -->
    <param name="width" value="$(arg width)"/>
    <param name="height" value="$(arg height)"/>
    <!-- choose whichever uncompressed format the camera supports: -->
    <param name="video_mode" value="$(arg video_mode)"/> <!-- or yuyv/nv12/mjpeg -->
    <param name="frame_rate" value="$(arg fps)"/>
    
    <param name="timestamp_method" value="$(arg timestamp_method)"/> <!-- start of frame -->
    <param name="camera_info_url" value="$(arg camera_info_url)"/>
    
    <param name="auto_exposure" value="$(arg auto_exposure)"/> <!-- use aperture_priority auto exposure -->
    <param name="auto_white_balance" value="$(arg auto_white_balance)"/>
    <param name="auto_focus" value="$(arg auto_focus)"/>
    <param name="exposure_absolute" value="$(arg exposure_absolute)"/>
    <param name="scanning_mode" value="$(arg scanning_mode)"/>
    <param name="auto_exposure_priority" value="$(arg auto_exposure_priority)"/>
  </node>
</launch>
